webservices/partition/base.py

```diff
@@ -142,186 +127,25 @@ def get_child_name(cls, cycle):
-                pass
-        create = utils.TableAs(name, select)
-        db.engine.execute(create)
-        child = utils.load_table(name)
-
-        cls.create_constraints(child, cycle, temp)
-        cls.create_indexes(child)
-        cls.update_child(child)
-        db.engine.execute(utils.Analyze(child))
-        logger.info(
-            'Successfully created child table {base}_{start}_{stop}.'.format(
-                base=cls.base_name,
-                start=start,
-                stop=stop
-            )
-        )
-        return child
-
-    @classmethod
-    def create_constraints(cls, child, cycle, temp=True):
-        start, stop = cycle - 1, cycle
-        master_name = '_tmp' if temp else ''
-        cmds = [
-            'alter table {child} alter column {primary} set not null',
-            'alter table {child} add primary key ({primary})',
-            'alter table {child} alter column filing_form set not null',
-            'alter table {child} add constraint check_two_year_transaction_period check (two_year_transaction_period in ({start}, {stop}))',  # noqa
-            'alter table {child} inherit {master}'
-        ]
-        params = {
-            'start': start,
-            'stop': stop,
-            'child': child.name,
-            'master': '{0}_master{1}'.format(cls.base_name, master_name),
-            'primary': cls.primary,
-        }
-        for cmd in cmds:
-            db.engine.execute(cmd.format(**params))
-
-    @classmethod
-    def create_indexes(cls, child):
-        for index in cls.index_factory(child):
-            try:
-                index.drop(db.engine)
-            except sa.exc.ProgrammingError:
-                pass
-            index.create(db.engine)
-
-    @classmethod
-    def rename(cls):
-        # Rename master table
-        cls.rename_master()
-
-        # Rename child tables
-        for cycle in get_cycles():
-            cls.rename_child(cycle)
-
-    @classmethod
-    def rename_master(cls):
-        cmds = [
-            'drop table if exists {0}_master cascade',
-            'alter table {0}_master_tmp rename to {0}_master',
-        ]
-
-        for cmd in cmds:
-            db.engine.execute(cmd.format(cls.base_name))
-
-    @classmethod
-    def rename_child(cls, cycle):
-        child_name = cls.get_child_name(cycle)
-        cmd = 'alter table {0}_tmp rename to {0}'.format(child_name)
-        db.engine.execute(cmd)
-        child = utils.load_table(child_name)
-
-        # Rename child table primary key
-        cmd = 'alter index {0} rename to {1}'.format(
-            child.primary_key.name,
-            child.primary_key.name.replace('_tmp', '')
-        )
-        db.engine.execute(cmd)
-
-        # Rename child table indexes
-        for index in child.indexes:
-            cmd = 'alter index {0} rename to {1}'.format(
-                index.name, index.name.replace('_tmp', '')
-            )
-            db.engine.execute(cmd)
```

Comment:
This code is what took care of renaming the indexes after all of the tables themselves were renamed - all it does is drop the `_tmp` part at the end of them, but it has to be done for each child table.